**L11 - Kajetan Walawski**

In [93]:
import csv
import pandas as pd

# Pobranie danych z pliku 
ulice = pd.read_csv('nyc_streets.csv')
ulice.head()

gid  id         name oneway           type
0    1   1  Shore Pky S    NaN    residential
1    2   2          NaN    NaN        footway
2    3   3     Avenue O    NaN    residential
3    4   4     Walsh Ct    NaN    residential
4    5   5          NaN    NaN  motorway_link

In [23]:
ulice.shape[0]

19091

**Odpowiedź na pytanie 1:**

Ilość rekordów w tabeli nyc_streets to 19091

In [11]:
ulice_2 = ulice.dropna(subset=['name'])

In [ ]:
ulice_B = ulice_2[ulice_2['name'].str.startswith('B')]
ulice_Q = ulice_2[ulice_2['name'].str.startswith('Q')]
ulice_M = ulice_2[ulice_2['name'].str.startswith('M')]

**Odpowiedź na pytanie 2:**

In [21]:
print('Ilość ulic rozpoczynających się na B:', ulice_B.shape[0])
print('Ilość ulic rozpoczynających się na Q:', ulice_Q.shape[0])
print('Ilość ulic rozpoczynających się na M:', ulice_M.shape[0])
print('Sumaryczna ilość ulic rozpoczynających się na B, Q lub M:', ulice_B.shape[0]+ulice_Q.shape[0]+ulice_M.shape[0])

Ilość ulic rozpoczynających się na B: 1282
Ilość ulic rozpoczynających się na Q: 68
Ilość ulic rozpoczynających się na M: 752
Sumaryczna ilość ulic rozpoczynających się na B, Q lub M: 2102


In [32]:
# Pobranie danych z pliku 
populacja = pd.read_csv('nyc_population.csv')
populacja.head()

gid            blkid  popn_total  popn_white  popn_black  popn_nativ  \
0    1  360850009001000          97          51          32           1   
1    2  360850020011000          66          52           2           0   
2    3  360850040001000          62          14          18           2   
3    4  360850074001000         137          92          12           0   
4    5  360850096011000         289         230           0           0   

   popn_asian  popn_other       boroname  
0           5           8  Staten Island  
1           7           5  Staten Island  
2          25           3  Staten Island  
3          13          20  Staten Island  
4          32          27  Staten Island

In [31]:
liczba = 0
for i in range(populacja.shape[0]):
    liczba += populacja['popn_total'][i]
liczba

8175032

**Odpowiedź na pytanie 3:**

Populacja miasta Nowy Jork wynosi: 8175032.

In [57]:
pop_bronx = populacja.loc[populacja['boroname'] == 'The Bronx']
pop_man = populacja.loc[populacja['boroname'] == 'Manhattan']
pop_queen = populacja.loc[populacja['boroname'] == 'Queens']

bronx = pop_bronx.sum(axis = 0, skipna = True)[2]
manhattan = pop_man.sum(axis = 0, skipna = True)[2]
queens = pop_queen.sum(axis = 0, skipna = True)[2]

**Odpowiedź na pytanie 4**

In [59]:
print('Populacja Bronxu wynosi: ', bronx)
print('Populacja Manhattanu wynosi: ', manhattan)
print('Populacja Queens wynosi: ', queens)
print('Populacja wszystkich trzech powyższych dzielnic wynosi: ', bronx+manhattan+queens)

Populacja Bronxu wynosi:  1385108
Populacja Manhattanu wynosi:  1585873
Populacja Queens wynosi:  2230621
Populacja wszystkich trzech powyższych dzielnic wynosi:  5201602


In [95]:
# Pobranie danych z pliku 
neighborhoods = pd.read_csv('nyc_neighborhoods.csv')
neighborhoods_2 = neighborhoods.drop(['gid'], axis=1)
neighborhoods_3 = neighborhoods_2.rename(columns={"boroname":"nazwa gminy", "name": "liczba dzielnic"})
result = neighborhoods_3.groupby(['nazwa gminy']).count()

**Odpowiedź na pytanie 5:**

In [94]:
result

liczba dzielnic
nazwa gminy                   
Brooklyn                    23
Manhattan                   28
Queens                      30
Staten Island               24
The Bronx                   24